### PR TITLE
Features 2267 feedback requests query parameters

### PR DESCRIPTION
**Pull request by mvolkmann**

### CODE REVIEW

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 1]
```diff
@@ -107,6 +107,35 @@ const ViewFeedbackPage = () => {
   const [dateRange, setDateRange] = useState(DateRange.THREE_MONTHS);
   const [includeAll, setIncludeAll] = useState(false);
 
+  useEffect(() => {
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:110]: On page load this sets all selection criteria from query parameters.

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 2]
```diff
@@ -107,6 +107,35 @@ const ViewFeedbackPage = () => {
...
+    setSearchText(searchText);
+
+    const showAll = url.searchParams.get('showAll');
+    setIncludeAll(showAll === 'true');
+
+    const sortValue = url.searchParams.get('sort') || SortOption.SENT_DATE;
+    setSortValue(sortValue);
+  }, []);
+
+  useEffect(() => {
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:126]: Every time one of the selection criteria changes, the URL is updated.

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 3]
```diff
@@ -380,16 +409,17 @@ const ViewFeedbackPage = () => {
                 </InputAdornment>
               )
             }}
+            value={searchText}
           />
-          <FormControl className={classes.textField} value={dateRange}>
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:384]: We don't need to set a value here. It is set in the TextField that this wraps.

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 4]
```diff
@@ -380,16 +409,17 @@ const ViewFeedbackPage = () => {
...
+          <FormControl className={classes.textField}>
             <TextField
               id="select-time"
               select
               fullWidth
               size="small"
               label="Show requests sent within"
               onChange={e => setDateRange(e.target.value)}
-              defaultValue={DateRange.THREE_MONTHS}
+              value={dateRange}
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:422]: We don't want to always default to THREE_MONTHS, but instead want to use the value in the corresponding query parameter.

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 5]
```diff
@@ -399,15 +429,15 @@ const ViewFeedbackPage = () => {
...
-          <FormControl className={classes.textField} value={sortValue}>
+          <FormControl className={classes.textField}>
             <TextField
               id="select-sort-method"
               select
               fullWidth
               size="small"
               label="Sort by"
               onChange={e => setSortValue(e.target.value)}
-              defaultValue={SortOption.SENT_DATE}
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:410]: We don't want to always default to SENT_DATE, but instead want to use the value in the corresponding query parameter.

File: web-ui/src/pages/ViewFeedbackPage.jsx [hunk 6]
```diff
@@ -399,15 +429,15 @@ const ViewFeedbackPage = () => {
             </TextField>
           </FormControl>
 
-          <FormControl className={classes.textField} value={sortValue}>
```
Comment (mvolkmann) [inline web-ui/src/pages/ViewFeedbackPage.jsx:402]: We don't need to set a value here. It is set in the TextField that this wraps.

**Review by S78901 — DISMISSED** [review]

**Review by vhscom — APPROVED** [review]

**Review by S78901 — APPROVED** [review]

**Pull request merged**